xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -900,7 +900,10 @@ bool CGUIMediaWindow::OnClick(int iItem, const std::string &player)
     else if (!CProfilesManager::GetInstance().GetCurrentProfile().canWriteSources() && !g_passwordManager.IsProfileLockUnlocked())
       return false;
 
-    return OnAddMediaSource();
+    if (OnAddMediaSource())
+      Refresh();
+
+    return true;
```

Comment:
Wondering if that should be `Refresh(true)` to clear the cache too.

When I had problems with the GUI not updating when adding a new music source it was cache related. I had been debugging, so the list got cached because of time taken, less likely in real life, but could happen with a server delay I guess. Anyway once cached that cache was never cleared or updated despite `GUI_MSG_UPDATE_SOURCES`  causing a refesh of GUIMediaWindow